ipam/calico-ipam.go

```diff
@@ -222,18 +222,34 @@ func cmdDel(args *skel.CmdArgs) error {
 		return err
 	}
 
-	// Release the IP address by using the handle - which is workloadID.
-	workloadID, _, err := utils.GetIdentifiers(args)
+	// Release the IP address by using the handleID.
+	handleId, err := utils.GetHandleId(args)
 	if err != nil {
 		return err
 	}
 
-	logger := utils.CreateContextLogger(workloadID)
+	// Release the IP address by using the workloadID, to not leak IPs in existing installs.
+	workloadId, _, err := utils.GetIdentifiers(args)
+	if err != nil {
+		return err
+	}
+
+	logger := utils.CreateContextLogger(handleId)
+
+	logger.Info("Releasing address using handleID")
+	if err := calicoClient.IPAM().ReleaseByHandle(handleId); err != nil {
+		if _, ok := err.(errors.ErrorResourceDoesNotExist); ok {
+			logger.WithField("handleId", handleId).Warn("Asked to release address but it doesn't exist. Ignoring")
+			return nil
+		}
+		return err
+	}
 
+	// We also need to handle the old way by workloadID, to not leak IPs in existing installs.
 	logger.Info("Releasing address using workloadID")
-	if err := calicoClient.IPAM().ReleaseByHandle(workloadID); err != nil {
+	if err := calicoClient.IPAM().ReleaseByHandle(workloadId); err != nil {
 		if _, ok := err.(errors.ErrorResourceDoesNotExist); ok {
-			logger.WithField("workloadId", workloadID).Warn("Asked to release address but it doesn't exist. Ignoring")
+			logger.WithField("workloadId", workloadId).Warn("Asked to release address but it doesn't exist. Ignoring")
```

Comment:
If we succeeded in deleting the key above, then we should suppress this warning.  Going forward, we only expect the above handle ID delete to succeed.